src/utils/hooks/api/loans/use-get-account-loans-overview.tsx

```diff
@@ -0,0 +1,180 @@
+import { useGetAccountPositions } from './use-get-account-positions';
+
+interface AccountLoansOverview {
+  lendPositions: LendPosition[] | undefined;
+  borrowPositions: BorrowPosition[] | undefined;
+  lentAssetsUSDValue: Big | undefined;
+  totalEarnedInterestUSDValue: Big | undefined;
+  borrowedAssetsUSDValue: Big | undefined;
+  borrowLimitUSDValue: Big | undefined;
+  collateralRatio: Big | undefined;
+  getNewCollateralRatio: (
+    type: 'withdraw' | 'deposit',
+    currency: CurrencyExt,
+    amount: MonetaryAmount<CurrencyExt>
+  ) => Big | undefined;
+  getNewBorrowLimitUSDValue: (
+    type: 'withdraw' | 'deposit',
+    currency: CurrencyExt,
+    amount: MonetaryAmount<CurrencyExt>
+  ) => Big | undefined;
+  getMaxBorrowableAmount: (
+    currency: CurrencyExt,
+    availableCapacity: MonetaryAmount<CurrencyExt>
+  ) => MonetaryAmount<CurrencyExt> | undefined;
+}
+
+const useGetAccountLoansOverview = (): AccountLoansOverview => {
+  const { selectedAccount } = useSubstrateSecureState();
+
+  const accountId = selectedAccount && newAccountId(window.bridge.api, selectedAccount.address);
+  const prices = useGetPrices();
+  const { lendPositions, borrowPositions } = useGetAccountPositions(accountId);
+
+  let lentAssetsUSDValue: Big | undefined = undefined;
+  let totalEarnedInterestUSDValue: Big | undefined = undefined;
+  let borrowedAssetsUSDValue: Big | undefined = undefined;
+  let collateralAssetsUSDValue: Big | undefined = undefined;
+  let collateralRatio: Big | undefined = undefined;
+
+  if (lendPositions !== undefined && prices !== undefined) {
+    lentAssetsUSDValue = getTotalUSDValueOfPositions(lendPositions, prices);
+    totalEarnedInterestUSDValue = getTotalEarnedInterestUSDValue(lendPositions, prices);
+
+    const collateralLendPositions = lendPositions.filter(({ isCollateral }) => isCollateral);
+    collateralAssetsUSDValue = getTotalUSDValueOfPositions(collateralLendPositions, prices);
+  }
+
+  if (borrowPositions !== undefined && prices !== undefined) {
+    borrowedAssetsUSDValue = getTotalUSDValueOfPositions(borrowPositions, prices);
+  }
+
+  if (borrowedAssetsUSDValue !== undefined && collateralAssetsUSDValue !== undefined) {
+    collateralRatio = collateralAssetsUSDValue.gt(0) ? borrowedAssetsUSDValue.div(collateralAssetsUSDValue) : Big(0);
+  }
+
+  const borrowLimitUSDValue = collateralAssetsUSDValue
+    ? collateralAssetsUSDValue.sub(borrowedAssetsUSDValue || 0)
+    : Big(0);
+
+  /**
+   * This method computes how the collateral ratio will change if
+   * asset is withdrawn or deposited.
+   * @param type Type of transaction that will be done: 'deposit' or 'withdraw'.
+   * @param currency Currency which will be deposited or withdrawn.
+   * @param amount Amount of `currency` that will be used.
+   * @note Call only after the prices and positions are loaded.
+   * @returns New collateral ratio after the transaction is done.
+   */
+  const getNewCollateralRatio = useCallback(
```

Comment:
Also, all these functions wrapped around `useCallback`'s, could just go out of the hook. This would make it smaller and easier to maintain.

Reply:
These callbacks are dependent on values owned by the hook, if we move them out then we would need to read and pass all of these values as parameters which would make the code more cluttered than needed. 

Can you share example of how you imagined that please?

Reply:
Sorry, I kinda missed the point that you are return this with the hook. Apologies. All good here.